python/tests/transformers/named_image_test.py

```diff
@@ -240,3 +242,31 @@ def test_featurizer_in_pipeline(self):
         pred_df_collected = lrModel.transform(train_df).collect()
         for row in pred_df_collected:
             self.assertEqual(int(row.prediction), row.label)
+
+
+class DeepImageFeaturizerPersistenceTest(SparkDLTestCase):
+    def test_inception(self):
+        transformer0 = DeepImageFeaturizer(inputCol='image', modelName="InceptionV3",
+                                          outputCol="features0", scaleHint="SCALE_FAST")
+        path = tempfile.mkdtemp()
+        dst_path = path + "/featurizer"
+        transformer0.save(dst_path)
+        transformer1 = DeepImageFeaturizer.load(dst_path)
+        self.assertEqual(transformer0.uid, transformer1.uid)
+        self.assertEqual(type(transformer0.uid), type(transformer1.uid))
+        self.assertEqual(transformer1.uid, transformer1.scaleHint.parent,
+                         "Loaded DeepImageFeaturizer instance uid (%s) did not match Param's uid (%s)"
+                         % (transformer1.uid, transformer1.scaleHint.parent))
```

Comment:
I'd check some other param since `scaleHint` has a default value. This test doesn't make sure other params are correctly read in.

Reply:
Sure, I can check uids on other params as well, but I don't think default value changes uid. As for the value, I check the whole paramMap.